system-addon/content-src/components/Sections/Sections.jsx

```diff
@@ -186,13 +189,35 @@ Section.defaultProps = {
 export const SectionIntl = injectIntl(Section);
 
 export class _Sections extends React.PureComponent {
+  renderSections() {
+    const sections = [];
+    const enabledSections = this.props.Sections.filter(section => section.enabled);
+    const {sectionOrder, showTopSites} = this.props.Prefs.values;
+    // Enabled sections doesn't include Top Sites, so we add it if enabled.
+    const expectedCount = enabledSections.length + ~~showTopSites;
+
+    for (const sectionId of sectionOrder.split(",")) {
+      const commonProps = {
+        key: sectionId,
+        isFirst: sections.length === 0,
+        isLast: sections.length === expectedCount - 1
+      };
+      if (sectionId === "topsites" && showTopSites) {
+        sections.push(<TopSites {...commonProps} />);
+      } else {
+        const section = enabledSections.find(s => s.id === sectionId);
+        if (section) {
+          sections.push(<SectionIntl {...section} Prefs={this.props.Prefs} dispatch={this.props.dispatch} {...commonProps} />);
```

Comment:
Personally, I would just connect `SectionIntl` to redux instead of passing in `Prefs` and `dispatch`